### PR TITLE
Convert docs from Sphinx to Hugo

### LINKED ISSUE #1225: Docs build is failing
Readthedocs is failing at this step:

```
python setup.py install --force
python: can't open file 'setup.py': [Errno 2] No such file or directory
```

### PR DESCRIPTION
**Pull request by jaylinski**
## Description

This PR converts the docs from Sphinx to Hugo (https://gohugo.io/).
The Sphinx build was broken for some time now (#1225) and the issue
is obviously not easy to track down. I chose Hugo because it has the advantage
of only needing a single binary, which makes building the docs very easy.

In order to build and publish the docs, a GitHub Action
was added. This action pushes the docs to the `gh-pages`
branch so it can be served by GitHub Pages. An example 
can be viewed here: https://github.com/jaylinski/phpspec/runs/431772750

The file-format was changed from rst to markdown (converted with pandoc), because it is a format most devs are familiar with.

A preview of the converted docs can be viewed here: https://jaylinski.github.io/phpspec/

### Positive effects

* The docs website is now ten times smaller and three times faster:
   New: `5 requests / 28.90 KB / 25.88 KB transferred / Finish: 459 ms / DOMContentLoaded: 182 ms / load: 460 ms`
   Old: `31 requests / 435.32 kB / 258.37 kB transferred / Finish: 1.84 s / DOMContentLoaded: 828 ms / load: 1.56 s`
* The docs website is now responsive (mobile-first)
* Build is super fast (210 ms for Hugo build, 24 seconds for complete pipeline) 
* 3,108 additions and 6,836 deletions (less code for same result)
* Adds an "Improve this page" button to the end of the docs
* Closes #1225 

### Downsides

* No more version selector
* No more readthedocs analytics
* No fancy icons on code highlighting boxes
* Menu item "phpspec manual" is not shown as active when a cookbook page is viewed (works only for pages in "manual" folder)
* No in-site search (you have to use DuckDuckGo or Google)
* Links are changing (although Hugo supports aliases/redirects)

## What do I as a maintainer have to do to get this to work?

1. Create a GitHub secret as described in https://github.com/peaceiris/actions-gh-pages#1-add-ssh-deploy-key
2. Merge this PR
3. Point the `phpspec.net` URL to the GitHub page

An alternative is deploying the docs to Netlify, which has the benefit of getting a free https certificate for the domain.

## What was my motivation to do this?

I wanted to try out GitHub Actions and saw this as a fitting opportunity. In hindsight I should have probably picked a static site generator specifically made for documentation, like GitBook, VuePress, Sculpin or some in https://github.com/myles/awesome-static-generators#documentation.

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Docs
+
+on: [push]  # TODO Change back to only master
```
Comment (jaylinski) [inline .github/workflows/docs.yml]: This should be changed (in case it gets merged).

File: docs/config.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+baseURL = "https://jaylinski.github.io/phpspec/" # TODO Change to phpspec.net
```
Comment (jaylinski) [inline docs/config.toml]: This needs to be changed (in case it gets merged).

**Comment by drupol** [comment]
So sad it's closed :(

**Comment by ciaranmcnulty** [comment]
It's because it targeted `master` which I removed, if @jaylinski can reopen against `main` I'm still interested

(It's mostly lack of Hugo knowledge stopping me know what to do with it)

**Comment by jaylinski** [comment]
@ciaranmcnulty I reopened the MR against `main: #1347. If you need any help regarding Hugo, feel free to ask.

**Pull request closed without merging**